ios-xcframework/.gitignore

```diff
@@ -6,3 +6,7 @@ Pods/
 
 # Ruby tooling
 vendor/bundle/
+
+# XCFramework build process
+archives
+xcframeworks
```

Comment:
Nitpick: We could put all these directories (`DerivedData`, `archives`, and `xcframeworks`) into a `build` directory? So that we just have one directory to ignore, and everything later related to "build" can be put into this `build` directory.

Reply:
Good point. It'll also simplify making local clean builds.

Addressed in 474592bf